File: src/main/java/gg/mineral/practice/commands/spectator/FollowCommand.java
package gg.mineral.practice.commands.spectator;

import gg.mineral.practice.commands.PlayerCommand;

import gg.mineral.practice.managers.ProfileManager;
import gg.mineral.practice.util.messages.impl.ChatMessages;
import gg.mineral.practice.util.messages.impl.ErrorMessages;
import gg.mineral.practice.util.messages.impl.UsageMessages;
import lombok.val;

public class FollowCommand extends PlayerCommand {

	public FollowCommand() {
		super("follow");
	}

	@Override
	public void execute(org.bukkit.entity.Player pl, String[] args) {
		val profile = ProfileManager.getOrCreateProfile(pl);
		if (args.length < 1) {
			profile.message(UsageMessages.FOLLOW);
			return;
		}

		if (profile.getName().equalsIgnoreCase(args[0])) {
			profile.message(ErrorMessages.NOT_FOLLOW_SELF);
			return;
		}

		val playerName = args[0];
		val profileToFollow = ProfileManager.getProfile(playerName);

		if (profileToFollow == null) {
			profile.message(ErrorMessages.PLAYER_NOT_ONLINE);
			return;
		}

		profile.getSpectateHandler().follow(profileToFollow);
		ChatMessages.FOLLOWING.clone().replace("%player%", playerName).send(pl);
	}
}

File: src/main/kotlin/gg/mineral/practice/commands/spectator/FollowCommand.kt
package gg.mineral.practice.commands.spectator

import dev.rollczi.litecommands.annotations.argument.Arg
import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.PlayerStatus
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.util.messages.impl.ChatMessages
import gg.mineral.practice.util.messages.impl.ErrorMessages

@Command(name = "follow")
class FollowCommand {
    @Execute
    fun execute(@Context profile: Profile, @Arg profileToFollow: Profile) {
        if (profile == profileToFollow) return profile.message(ErrorMessages.NOT_FOLLOW_SELF)
        if (profile.playerStatus !== PlayerStatus.IDLE) return profile.message(ErrorMessages.YOU_ARE_NOT_IN_LOBBY)

        profile.following = profileToFollow
        profile.message(ChatMessages.FOLLOWING.clone().replace("%player%", profileToFollow.name))
    }
}
